iothub/device/samples/convention-based-samples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,375 @@
+
+                return Task.FromResult(new CommandResponse(StatusCodes.NotFound));
+            }
+            catch (JsonReaderException ex)
+            {
+                _logger.LogError($"Command input for {commandRequest.CommandName} is invalid: {ex.Message}.");
+
+                return Task.FromResult(new CommandResponse(StatusCodes.BadRequest));
+            }
+        }
+
+        // Report the property updates on "deviceInformation" component.
+        private async Task UpdateDeviceInformationAsync(CancellationToken cancellationToken)
+        {
+            const string componentName = "deviceInformation";
+            var deviceInformationProperties = new Dictionary<string, object>
+            {
+                { "manufacturer", "element15" },
+                { "model", "ModelIDxcdvmk" },
+                { "swVersion", "1.0.0" },
+                { "osName", "Windows 10" },
+                { "processorArchitecture", "64-bit" },
+                { "processorManufacturer", "Intel" },
+                { "totalStorage", 256 },
+                { "totalMemory", 1024 },
+            };
+            var deviceInformation = new ClientPropertyCollection();
+            deviceInformation.Add(componentName, deviceInformationProperties);
+
+            ClientPropertiesUpdateResponse updateResponse = await _deviceClient.UpdateClientPropertiesAsync(deviceInformation, cancellationToken);
+
+            _logger.LogDebug($"Property: Update - component = '{componentName}', properties update is complete " +
+                $"with a version of {updateResponse.Version}.");
+        }
+
+        // Send working set of device memory over telemetry.
+        private async Task SendDeviceMemoryAsync(CancellationToken cancellationToken)
+        {
+            const string workingSetName = "workingSet";
+            long workingSet = Process.GetCurrentProcess().PrivateMemorySize64 / 1024;
+            using var telemetryMessage = new TelemetryMessage
+            {
+                Telemetry = { [workingSetName] = workingSet },
+            };
+
+            await _deviceClient.SendTelemetryAsync(telemetryMessage, cancellationToken);
+
+            _logger.LogDebug($"Telemetry: Sent - {telemetryMessage.Telemetry.GetSerializedString()} in KB.");
+        }
+
+        // Send device serial number over property update.
+        private async Task SendDeviceSerialNumberAsync(CancellationToken cancellationToken)
+        {
+            const string propertyName = "serialNumber";
+            const string serialNumber = "SR-123456";
+
+            var reportedProperties = new ClientPropertyCollection();
+            reportedProperties.Add(propertyName, serialNumber);
+
+            ClientPropertiesUpdateResponse updateResponse = await _deviceClient.UpdateClientPropertiesAsync(reportedProperties, cancellationToken);
+
+            _logger.LogDebug($"Property: Update - {{ {reportedProperties.GetSerializedString()} }} is complete " +
```

Comment:
What is the reason for the double curly braces? It looks like you are escaping them, but it makes the formatable string hard to read. Can we just go with a different text format?

Reply:
Yes, the intension is to escape them, is there a different format that can be used for braces when using string interpolation?

Reply:
I'm a fan of square brackets. You could use quotation marks. You could choose to surround it with nothing.